src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ProvideCorrectArgumentsToFormattingMethods.cs

```diff
@@ -355,6 +355,13 @@ public StringFormatInfo(Compilation compilation)
                     return info;
                 }
 
+                // Check if method could be a string formatting method.
+                // This is defined as a method with a 'format' argument followed by an 'params object[]'.
+                if (TryGetFormatInfo(method, out info))
```

Comment:
As far as I recall, this behavior was considered in the original implementation, but was not chosen due to following reasons:
1. Potential of being a breaking change for the rule.
2. Makes this a heuristic based rule. Current implementation of the rule is conservative to the known or user configured set of formatting methods. This makes it aggressive to fire on all methods which have similar signature.

Given that this rule is marked as `RuleLevel.BuildWarningCandidate`, I am reluctant to change the default behavior to make it so aggressive. At the minimum, we need an option to guard this new behavior so users can select the preferred level of aggressiveness.

@dotnet/roslyn-analysis for opinions.